wire-runtime/src/test/java/com/squareup/wire/ProtoAdapterTest.java

```diff
@@ -35,6 +35,18 @@
     assertThat(personAdapter.decode(encoded.toByteArray())).isEqualTo(person);
   }
 
+  @Test public void getFromInstance() throws Exception {
+    Person person = new Person.Builder()
+        .id(99)
+        .name("Omar Little")
+        .build();
+    ByteString encoded = ByteString.decodeHex("0a0b4f6d6172204c6974746c651063");
+
+    ProtoAdapter<Person> personAdapter = ProtoAdapter.get(person);
+    assertThat(ByteString.of(personAdapter.encode(person))).isEqualTo(encoded);
+    assertThat(personAdapter.decode(encoded.toByteArray())).isEqualTo(person);
```

Comment:
I think the only assertion you need is that the same instance as `Person.ADAPTER` was returned. That's the contract of the method and encoding is covered elsewhere.